lib/compress.js

```diff
@@ -1273,7 +1273,25 @@ merge(Compressor.prototype, {
         def(AST_Constant, return_false);
         def(AST_This, return_false);
 
+        function has_pure_annotation(node) {
+            if (node.pure !== undefined) return node.pure;
+            var pure = false;
+            var comments, last_comment;
+            if (node.start
+                && (comments = node.start.comments_before)
+                && comments.length
+                && /[@#]__PURE__/.test((last_comment = comments[comments.length - 1]).value)) {
+                last_comment.value = last_comment.value.replace(/[@#]__PURE__/g, ' ');
```

Comment:
@kzc any reason why the `#__PURE__` comment has to be the last one? why this isnt checking all `node.start.comments_before`?

If im gonna write i.e. a babel plugin which annotates calls for me it might interfere with other annotations causing i.e. such situation
```js
var A = /*#__PURE__*/ /* @class */ function() { /* ... */ }();
```
and thus preventing it from being dropped as unused.

Reply:
> any reason why the `#__PURE__` comment has to be the last one? why this isnt checking all `node.start.comments_before`?

No particular reason other than efficiency. It probably should check all comments. PR is welcome.

Be aware that it's unlikely to be back-ported to `uglify-js@2.x` which is widely used but whose code base is frozen. So you'd probably want to list `#__PURE__` in the last comment to aid legacy uglify users.

Reply:
> No particular reason other than efficiency. It probably should check all comments. PR is welcome.

PR is comming then. :)

> Be aware that it's unlikely to be back-ported to uglify-js@2.x which is widely used but whose code base is frozen. So you'd probably want to list #__PURE__ in the last comment to aid legacy uglify users.

While at the moment of adding the comment I can ensure its added as the last comment, I cannot be sure that no other transformation change it. I wouldnt be overly concern about old version of the uglify. Life goes on and library's consumers should upgrade sooner or later :)

Reply:
I should mention that the following would still work with current uglify versions:
```js
var A = /* @__PURE__ @class */ function() { /* ... */ }();
```
That's all the proposed babel plugin would have to do to retain backwards compatibility with uglify.